lms/djangoapps/badges/backends/badgr.py

```diff
@@ -0,0 +1,165 @@
+"""
+Badge Awarding backend for Badgr-Server.
+"""
+import hashlib
+import logging
+import mimetypes
+
+import requests
+from django.conf import settings
+from lazy import lazy
+from requests.packages.urllib3.exceptions import HTTPError
+
+from badges.backends.base import BadgeBackend
+from eventtracking import tracker
+
+from badges.models import BadgeAssertion
+
+MAX_SLUG_LENGTH = 255
+LOGGER = logging.getLogger(__name__)
+
+
+class BadgrBackend(BadgeBackend):
+    """
+    Backend for Badgr-Server by Concentric Sky. http://info.badgr.io/
+    """
+    badges = []
+
+    def __init__(self):
+        assert settings.BADGR_API_TOKEN
```

Comment:
I'd prefer a more specific exception here.  I've seen some parts of the platform are using `django.core.exceptions.ImproperlyConfigured`, which might be a good choice here as well.